internal/sync.go

```diff
@@ -223,25 +251,32 @@ func (s *syncGSuite) SyncGroups(query string) error {
 
 		for _, u := range s.users {
 			log.WithField("user", u.Username).Debug("Checking user is in group already")
-			b, err := s.aws.IsUserInGroup(u, group)
-			if err != nil {
-				return err
+			var b bool
+			if !testing {
+				b, err = s.aws.IsUserInGroup(u, group)
+				if err != nil {
+					return err
+				}
 			}
 
 			if _, ok := memberList[u.Username]; ok {
 				if !b {
 					log.WithField("user", u.Username).Info("Adding user to group")
-					err := s.aws.AddUserToGroup(u, group)
-					if err != nil {
-						return err
+					if !testing {
+						err := s.aws.AddUserToGroup(u, group)
+						if err != nil {
+							return err
+						}
 					}
 				}
 			} else {
 				if b {
 					log.WithField("user", u.Username).Warn("Removing user from group")
-					err := s.aws.RemoveUserFromGroup(u, group)
-					if err != nil {
-						return err
+					if !testing {
+						err := s.aws.RemoveUserFromGroup(u, group)
+						if err != nil {
+							return err
+						}
 					}
 				}
```

Comment:
If `testing` is `false`, then `b` will always be `false`, meaning this nested `if` could possibly mislead that there are times when `testing` and `b` aren't directly linked. Should the `if !testing` be removed (with a comment added) or a comment added in generally saying "if we're testing, this will never run, but keeping the testing statement for consistency"?

Reply:
Thanks!  Yeah I was thinking about this to and went with the later thought that I should probably keep the `if !testing` to maintain consistency.  But I like the thought of adding a comment.  I'll do that now.